lib/reservation_creator.rb

```diff
@@ -0,0 +1,26 @@
+require 'date'
+
+module Hotel
+  class ReservationCreator
```

Comment:
When I look at this class, it really feels like the attributes, state, behaviors that this class has resemble more of representing a Reservation.

If this object were a "ReservationCreator," I would imagine that its behaviors would be more about making Reservation instances, and wouldn't need a check in or check out date.